js/geo/transform.js

```diff
@@ -345,27 +336,17 @@ Transform.prototype = {
     /**
      * Given a coordinate, return the screen point that corresponds to it
      * @param {Coordinate} coord
-     * @returns {Point} screen point
+     * @param {boolean=false} shouldReturnRawVector
+     * @returns {Point} screen point (or 4d vector representing the transform if `shouldReturnRawVector` is true)
      * @private
      */
-    coordinatePoint: function(coord) {
+    coordinatePoint: function(coord, shouldReturnRawVector) {
         var scale = this.worldSize / this.zoomScale(coord.zoom);
         var p = [coord.column * scale, coord.row * scale, 0, 1];
         vec4.transformMat4(p, p, this.pixelMatrix);
-        return new Point(p[0] / p[3], p[1] / p[3]);
-    },
 
-    /**
-     * Like `coordinatePoint` but returns the raw vector data instead of just the Point
-     * @param {Coordinate} coord
-     * @returns {Point} screen point
-     * @private
-     */
-    coordinatePoint3d: function(coord) {
-        var scale = this.worldSize / this.zoomScale(coord.zoom);
-        var p = [coord.column * scale, coord.row * scale, 0, 1];
-        vec4.transformMat4(p, p, this.pixelMatrix);
-        return p;
+        if(shouldReturnRawVector) return p;
+        return new Point(p[0] / p[3], p[1] / p[3]);
```

Comment:
👎  on changing the return type based on an argument. Two seperate methods are preferable. 

It looks like this code could be written most succinctly if `coordiantePoint` called `coordinatePoint3d`

```
coordinatePoint: function() {
    var p = this.coordinatePoint3d(coord);
    return new Point(p[0] / p[3], p[1] / p[3]);
}
```